### ISSUE TITLE
Alabama scraper broken

### ISSUE THREAD

**Issue by brianwc** [freelawproject/juriscraper#46]
I wrote a long description and @github mobile erased it when I slid my finger wrong way. Ugh. Little warning?
See:
http://judicial.alabama.gov/supreme_opinions.cfm

Opinions are missing. Important one today.

**Comment by mlissner** [comment]
It appears Alalinc is gone. This issue may not be solvable and we may have to drop support for Alabama. The only way to get their opinions now is by parsing a PDF, a frontier we've successfully avoided until now...

**Comment by mlissner** [comment]
Well, we've done some early investigation into scraping PDFs, but I don't have enough time to keep working on this atm, so here are my notes:

1. The first approach I tried was to use [`pdfquery`][pdfq]. The big problem I ran into there was that it doesn't seem to preserve links inside PDFs. I [filed an issue about this][issue], so perhaps a solution will be found. If none is found, `pdfquery` is a dead end.

2. @brianwc did an experiment with using pdftohtml to convert the pdf to html that we can then scrape. I think this is a good way forward and that the likely solution here is to override Juriscraper's `_parse` or `_download` method to:

 1. Download the PDF that contains the index of cases.
 1. Convert it to HTML using `pdftohtml`.
 1. Put that HTML in memory and then hand it over to the scraper as the source.

 I think this strategy will actually work quite well, but I it's an area that'll need more research, since we haven't done it before. I'd welcome anybody to try to sort this out.

For now, I have to dedicate my efforts elsewhere and Alamaba is *OFFLINE*. Please help if you can.

[pdfq]: https://github.com/jcushman/pdfquery
[issue]: https://github.com/jcushman/pdfquery/issues/28

**Comment by mlissner** [comment]
Update: I fixed the issue in pdfquery, so this should be unblocked.

**Comment by arderyp** [comment]
sorry I'm chiming in on this late, but, given your last comment @mlissner about pdfquery, does that mean you'd rather move forward with that, or stick to the pdftohtml plan?  I do REALLY like the idea of simply converting a PDF to html and handing it off to the parser, as it seems like an elegant way of leveraging the current scraper framework to get the job done with minimal extra coding.  That being said, I've had very little experience with pdftohtml and suspect that it could get hairy (like any pdf parsing utility) if the court is publishing documents with quarky/weird/nonstandard/bad formatting.

**Comment by mlissner** [comment]
I forget why exactly pdfquery was better than pdftohtml, but I do recall that being the case. I think fixing this court is pretty low priority though. Their site is so broken the same amount of effort could probably fix two or three others.

**Comment by flooie** [comment]
is this still open?

**Comment by arderyp** [comment]
Haha I have no idea.  I doubt it, but I will test alabama when I get home.

**Comment by arderyp** [comment]
@flooie there is no alabama scraper anymore, so I am closing this.  @mlissner, I don't remember the story here, but if you think this needs to be re-opened or need us to look into something, just let us know.

For what its worth, their method of publishing opinions is via pdf text documents, so I am not sure we could start scraping them again without pdf parsing or an alternate data source, though [justia is doing it somehow](https://law.justia.com/cases/alabama/supreme-court/):

http://judicial.alabama.gov/decision/supremecourtdecisions

They also have one of the most hilariously awful side navigation menus I've ever seen.

**Comment by flooie** [comment]
Fascinating.  Would it be okay with you to take a shot at some point?  @arderyp

**Comment by brianwc** [comment]
Oh, I don't remember which state it was, but I think we previously dealt
with one that listed all their opinions via links within a PDF. So, I think
we solved this once before. Perhaps @mlissner recalls.

**Comment by arderyp** [comment]
@flooie, go for it, and @brianwc, perhaps you are correct.  If so, we should probably just open a new issue to create a new scraper, since the actual issue here (broken scraper) is no longer relevant.

**Comment by mlissner** [comment]
I think Colorado had a site where they were doing the PDF thing, but their bar association had a better site, so we started using that? I forget, but in any case, we never dealt with one of these PDF parsing ones before. 

I guess it's doable, but it's going to be tricky to get right thanks to needing to get metadata out of a PDF, and it's going to be difficult to integrate into our regular scraping system.

**Comment by mlissner** [comment]
Oh, one other thing. If we're really going to go after this, we should see if they have any other ways of getting the data or if they'd be willing to work with us to post things in a better way on their website. Why square a circle if we can have a circle in the first place via some diplomacy.

**Comment by flooie** [comment]
I have a fix for this incoming.

**Comment by flooie** [comment]
<img width="1042" alt="Screenshot 2023-01-11 at 8 44 59 PM" src="https://user-images.githubusercontent.com/6464529/211956158-af23ce73-d411-458b-a503-e55705c25c61.png">
